internal/builders/docker/pkg/config.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkg
+
+// This file contains functionality and structs for validating and
+// representing user inputs and configuration files.
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	toml "github.com/pelletier/go-toml"
+	"github.com/slsa-framework/slsa-github-generator/internal/utils"
+)
+
+// BuildConfig is a collection of parameters to use for building the artifact.
+type BuildConfig struct {
+	// TODO(#1191): Add env and options if needed.
+	// Command to pass to `docker run`. The command is taken as an array
+	// instead of a single string to avoid unnecessary parsing. See
+	// https://docs.docker.com/engine/reference/builder/#cmd and
+	// https://man7.org/linux/man-pages/man3/exec.3.html for more details.
+	Command []string `toml:"command"`
+
+	// The path, relative to the root of the git repository, where the artifact
+	// built by the `docker run` command is expected to be found.
+	ArtifactPath string `toml:"artifact_path"`
+}
+
+// Digest specifies a digest values, including the name of the hash function
+// that was used for computing the digest.
+type Digest struct {
+	Alg   string
+	Value string
+}
+
+// DockerImage fully specifies a docker image by a URI (e.g., including the
+// docker image name and registry), and its digest.
+type DockerImage struct {
+	URI    string
+	Digest Digest
+}
+
+// ToString returns the builder image in the form of NAME@ALG:VALUE.
+func (bi *DockerImage) ToString() string {
+	return fmt.Sprintf("%s@%s:%s", bi.URI, bi.Digest.Alg, bi.Digest.Value)
+}
+
+// DockerBuildConfig is a convenience class for holding validated user inputs.
+type DockerBuildConfig struct {
+	SourceRepo      string
+	SourceDigest    Digest
+	BuilderImage    DockerImage
+	BuildConfigPath string
+}
+
+// NewDockerBuildConfig validates the inputs and generates an instance of
+// DockerBuildConfig.
+func NewDockerBuildConfig(io *InputOptions) (*DockerBuildConfig, error) {
```

Comment:
I could make the fields in `DockerBuildConfig` private/unexposed to make sure it is not possible to create an instance of it without performing these checks. What do you think? @asraa & @ianlewis (somewhat related to the comment below on validating the path)